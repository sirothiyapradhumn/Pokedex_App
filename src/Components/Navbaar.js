import React, { useState } from 'react'


const types = ['Grass','Poison','Fire','Flying','Water','Bug','Normal','Electric','Ground','Psychic','Rock','Ice','Ghost','Dragon','Fighting'];

  const weakness = [
    'Grass',
    'Poison',
    'Fire',
    'Flying',
    'Water',
    'Bug',
    'Fairy',
    'Electric',
    'Ground',
    'Psychic',
    'Rock',
    'Ice',
    'Ghost',
    'Dragon',
    'Fighting',
    'Steel',
    'Dark',
  ];

function Navbaar() {
    const [checkedStateTyp, setCheckedStateTyp] = useState(new Array(types.length).fill(false));
    //console.log(checkedStateTyp);
    const [checkedStateWeak, setCheckedStateWeak] = useState(new Array(weakness.length).fill(false));
    const [pokeType, setpokeType] = useState([]);
    //console.log(pokeType);
    const [pokeWeakness, setPokeWeakness] = useState([]);
    console.log(pokeWeakness);
    
    const handleOnChange = (position) => {
        const updatedCheckedState = checkedStateTyp.map((item, index) =>
          index === position ? !item : item
        );

        setCheckedStateTyp(updatedCheckedState);

        let tempData = updatedCheckedState.reduce((arr, currState, idx) =>{
            if(currState === true){
                arr.push(types[idx])
            }
            return arr;
        },[]);
        
        setpokeType(tempData);
    }

    const handleOnChange1 = (position) => {
        const updatedCheckedState = checkedStateWeak.map((item, index) =>
          index === position ? !item : item
        );

        setCheckedStateWeak(updatedCheckedState);

        let tempData = updatedCheckedState.reduce((arr, currState, idx) =>{
            if(currState === true){
                arr.push(types[idx])
            }
            return arr;
        },[]);
        
        setPokeWeakness(tempData);
    }

  return (
    <div className='navbar-self'>
        <div className='title'>
            <h1>Pokedex</h1>
        </div>

        <div className='poke-types'>
            <h3>Types</h3>
            <ul className="list">
            {types.map((type, index) => {
                return (
                    <li key={index}>
                    <div className="toppings-list-item">
                        <div className="left-section">
                        <input 
                            type="checkbox" name={type} 
                            value={type} 
                            checked = {checkedStateTyp[index]}
                            onChange={()=>handleOnChange(index)} 
                            />

                        <label>{type}</label>
                        </div>
                    </div>
                    </li>
                );
            })}
            </ul>
        </div>

        <div className='poke-weekness'>
            <h3>Weaknesses</h3>
            <ul className="list">
            {weakness.map((week, index) => {
                return (
                    <li key={index}>
                    <div className="toppings-list-item">
                        <div className="left-section">
                        <input type="checkbox" name={week} value={week} 
                        checked = {checkedStateWeak[index]}
                        onChange={()=>handleOnChange1(index)} 
                        />
                        <label>{week}</label>
                        </div>
                    </div>
                    </li>
                );
            })} 
            </ul>           
        </div>

        <div className='btn-go'>
            <button>GO</button>
        </div>

        <div className='search-bar'>
            <input type="text" placeholder="Search.."></input>
        </div>
    </div>
  )
}

export default Navbaar