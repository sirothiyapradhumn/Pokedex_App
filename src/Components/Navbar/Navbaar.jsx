import React, { useState } from 'react'
import List from '../List/List';
import style from './navbaar.module.scss'

const types = ['Grass','Poison','Fire','Flying','Water','Bug','Normal','Electric','Ground','Psychic','Rock','Ice','Ghost','Dragon','Fighting'];

  const weakness = [
    'Grass',
    'Poison',
    'Fire',
    'Flying',
    'Water',
    'Bug',
    'Fairy',
    'Electric',
    'Ground',
    'Psychic',
    'Rock',
    'Ice',
    'Ghost',
    'Dragon',
    'Fighting',
    'Steel',
    'Dark',
  ];

function Navbaar(props) {
    const [checkedStateTyp, setCheckedStateTyp] = useState(new Array(types.length).fill(false));
    const [checkedStateWeak, setCheckedStateWeak] = useState(new Array(weakness.length).fill(false));
    const [pokeType, setpokeType] = useState([]);
    const [pokeWeakness, setPokeWeakness] = useState([]);
    const [currText, setCurrText] = useState('');

    const handleOnChange = (position) => {
        const updatedCheckedState = checkedStateTyp.map((item, index) =>
            index === position ? !item : item
        );

        setCheckedStateTyp(updatedCheckedState);

        let tempData = updatedCheckedState.reduce((arr, currState, idx) =>{
            if(currState === true){
                arr.push(types[idx])
            }
            return arr;
        },[]);
        
        setpokeType(tempData);
    }

    const handleOnChange1 = (position) => {
        const updatedCheckedState = checkedStateWeak.map((item, index) =>
          index === position ? !item : item
        );

        setCheckedStateWeak(updatedCheckedState);

        let tempData = updatedCheckedState.reduce((arr, currState, idx) =>{
            if(currState === true){
                arr.push(types[idx])
            }
            return arr;
        },[]);
        
        setPokeWeakness(tempData);
    }

return (
    <>
    <div className={style.navbarSelf}>
        <div className={style.title}>
            <h1 style={{color: "#66fcf1",margin: 0, fontWeight: "900",fontSize: "3rem"}}>Pokedex</h1>
        </div>

        <div className={style.pokeTypes}>
            <h3 style={{color: "#66fcf1"}}>Types</h3>
            <ul className={style.list}>
            {types.map((type, index) => {
                return (
                    <li key={index}>
                    <div>
                        <div>
                        <input 
                            type="checkbox" name={type} 
                            value={type} 
                            checked = {checkedStateTyp[index]}
                            onChange={()=>handleOnChange(index)} 
                            />

                        <label style={{color: "#66fcf1"}}>{type}</label>
                        </div>
                    </div>
                    </li>
                );
            })}
            </ul>
        </div>

        <div className={style.pokeWeekness}>
            <h3 style={{color: "#66fcf1"}}>Weaknesses</h3>
            <ul className={style.list}>
            {weakness.map((week, index) => {
                return (
                    <li key={index}>
                    <div>
                        <div>
                        <input type="checkbox" name={week} value={week} 
                        checked = {checkedStateWeak[index]}
                        onChange={()=>handleOnChange1(index)} 
                        />
                        <label style={{color: "#66fcf1"}}>{week}</label>
                        </div>
                    </div>
                    </li>
                );
            })} 
            </ul>           
        </div>
        <div className={style.searchBar}>
            <input type="text" placeholder="Search.." value={currText} onChange={(e)=> setCurrText(e.target.value)} />
        </div>
    </div>
    <List currText={currText} pokeType={pokeType} pokeWeakness={pokeWeakness} data={props.data} apiCall={props.apiCall} />
    </>
  )
}

export default Navbaar