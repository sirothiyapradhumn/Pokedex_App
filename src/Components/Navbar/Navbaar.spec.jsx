import { mount, shallow } from 'enzyme';
import Navbaar from './Navbaar'

describe('Navbar Components', ()=>{

  it('check navbarself class', ()=>{
      let wrapper = shallow(<Navbaar/>);
      expect(wrapper.find('.navbarSelf').length).toBe(1);
  })

  it('check h1 msg', ()=>{
    let wrapper = shallow(<Navbaar/>);
    expect(wrapper.contains(<h1 style={{color: "#66fcf1",margin: 0, fontWeight: "900",fontSize: "3rem"}}>Pokedex</h1>)).toBe(true);
})

  it('check all li count', ()=>{
    let wrapper = shallow(<Navbaar/>);
    expect(wrapper.find('li').length).toBe(32);
})

it('check all li count1', ()=>{
  let wrapper = shallow(<Navbaar/>);
  expect(wrapper.find('li').length).toBe(32);
})

it('defalut snapshot', ()=>{
  let wrapper = shallow(<Navbaar/>);
  expect(wrapper).toMatchSnapshot();
})

it('input first click snapshot', ()=>{
  let wrapper = shallow(<Navbaar/>);
  let input= wrapper.find('input').first();
  input.simulate('change', true);
  expect(input).toMatchSnapshot();
})

it('input last click snapshot', ()=>{
  let wrapper = shallow(<Navbaar/>);
  let input= wrapper.find({ value : "Steel" });
  input.simulate('change', true);
  expect(input).toMatchSnapshot();
})

it('input search snapshot', ()=>{
  let wrapper = shallow(<Navbaar/>);
  const event = {target: {value: "blub"}};
  let input= wrapper.find('input').last();
  input.simulate('change', event)
  expect(input).toMatchSnapshot();
})

});
