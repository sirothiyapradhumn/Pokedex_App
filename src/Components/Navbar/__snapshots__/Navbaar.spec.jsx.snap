// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Components defalut snapshot 1`] = `
<Fragment>
  <div
    className="navbarSelf"
  >
    <div
      className="title"
    >
      <h1
        style={
          Object {
            "color": "#66fcf1",
            "fontSize": "3rem",
            "fontWeight": "900",
            "margin": 0,
          }
        }
      >
        Pokedex
      </h1>
    </div>
    <div
      className="pokeTypes"
    >
      <h3
        style={
          Object {
            "color": "#66fcf1",
          }
        }
      >
        Types
      </h3>
      <ul
        className="list"
      >
        <li
          key="0"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Grass"
                onChange={[Function]}
                type="checkbox"
                value="Grass"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Grass
              </label>
            </div>
          </div>
        </li>
        <li
          key="1"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Poison"
                onChange={[Function]}
                type="checkbox"
                value="Poison"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Poison
              </label>
            </div>
          </div>
        </li>
        <li
          key="2"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Fire"
                onChange={[Function]}
                type="checkbox"
                value="Fire"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Fire
              </label>
            </div>
          </div>
        </li>
        <li
          key="3"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Flying"
                onChange={[Function]}
                type="checkbox"
                value="Flying"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Flying
              </label>
            </div>
          </div>
        </li>
        <li
          key="4"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Water"
                onChange={[Function]}
                type="checkbox"
                value="Water"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Water
              </label>
            </div>
          </div>
        </li>
        <li
          key="5"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Bug"
                onChange={[Function]}
                type="checkbox"
                value="Bug"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Bug
              </label>
            </div>
          </div>
        </li>
        <li
          key="6"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Normal"
                onChange={[Function]}
                type="checkbox"
                value="Normal"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Normal
              </label>
            </div>
          </div>
        </li>
        <li
          key="7"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Electric"
                onChange={[Function]}
                type="checkbox"
                value="Electric"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Electric
              </label>
            </div>
          </div>
        </li>
        <li
          key="8"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ground"
                onChange={[Function]}
                type="checkbox"
                value="Ground"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ground
              </label>
            </div>
          </div>
        </li>
        <li
          key="9"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Psychic"
                onChange={[Function]}
                type="checkbox"
                value="Psychic"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Psychic
              </label>
            </div>
          </div>
        </li>
        <li
          key="10"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Rock"
                onChange={[Function]}
                type="checkbox"
                value="Rock"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Rock
              </label>
            </div>
          </div>
        </li>
        <li
          key="11"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ice"
                onChange={[Function]}
                type="checkbox"
                value="Ice"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ice
              </label>
            </div>
          </div>
        </li>
        <li
          key="12"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ghost"
                onChange={[Function]}
                type="checkbox"
                value="Ghost"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ghost
              </label>
            </div>
          </div>
        </li>
        <li
          key="13"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Dragon"
                onChange={[Function]}
                type="checkbox"
                value="Dragon"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Dragon
              </label>
            </div>
          </div>
        </li>
        <li
          key="14"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Fighting"
                onChange={[Function]}
                type="checkbox"
                value="Fighting"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Fighting
              </label>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="pokeWeekness"
    >
      <h3
        style={
          Object {
            "color": "#66fcf1",
          }
        }
      >
        Weaknesses
      </h3>
      <ul
        className="list"
      >
        <li
          key="0"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Grass"
                onChange={[Function]}
                type="checkbox"
                value="Grass"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Grass
              </label>
            </div>
          </div>
        </li>
        <li
          key="1"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Poison"
                onChange={[Function]}
                type="checkbox"
                value="Poison"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Poison
              </label>
            </div>
          </div>
        </li>
        <li
          key="2"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Fire"
                onChange={[Function]}
                type="checkbox"
                value="Fire"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Fire
              </label>
            </div>
          </div>
        </li>
        <li
          key="3"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Flying"
                onChange={[Function]}
                type="checkbox"
                value="Flying"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Flying
              </label>
            </div>
          </div>
        </li>
        <li
          key="4"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Water"
                onChange={[Function]}
                type="checkbox"
                value="Water"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Water
              </label>
            </div>
          </div>
        </li>
        <li
          key="5"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Bug"
                onChange={[Function]}
                type="checkbox"
                value="Bug"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Bug
              </label>
            </div>
          </div>
        </li>
        <li
          key="6"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Fairy"
                onChange={[Function]}
                type="checkbox"
                value="Fairy"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Fairy
              </label>
            </div>
          </div>
        </li>
        <li
          key="7"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Electric"
                onChange={[Function]}
                type="checkbox"
                value="Electric"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Electric
              </label>
            </div>
          </div>
        </li>
        <li
          key="8"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ground"
                onChange={[Function]}
                type="checkbox"
                value="Ground"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ground
              </label>
            </div>
          </div>
        </li>
        <li
          key="9"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Psychic"
                onChange={[Function]}
                type="checkbox"
                value="Psychic"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Psychic
              </label>
            </div>
          </div>
        </li>
        <li
          key="10"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Rock"
                onChange={[Function]}
                type="checkbox"
                value="Rock"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Rock
              </label>
            </div>
          </div>
        </li>
        <li
          key="11"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ice"
                onChange={[Function]}
                type="checkbox"
                value="Ice"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ice
              </label>
            </div>
          </div>
        </li>
        <li
          key="12"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Ghost"
                onChange={[Function]}
                type="checkbox"
                value="Ghost"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Ghost
              </label>
            </div>
          </div>
        </li>
        <li
          key="13"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Dragon"
                onChange={[Function]}
                type="checkbox"
                value="Dragon"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Dragon
              </label>
            </div>
          </div>
        </li>
        <li
          key="14"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Fighting"
                onChange={[Function]}
                type="checkbox"
                value="Fighting"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Fighting
              </label>
            </div>
          </div>
        </li>
        <li
          key="15"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Steel"
                onChange={[Function]}
                type="checkbox"
                value="Steel"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Steel
              </label>
            </div>
          </div>
        </li>
        <li
          key="16"
        >
          <div>
            <div>
              <input
                checked={false}
                name="Dark"
                onChange={[Function]}
                type="checkbox"
                value="Dark"
              />
              <label
                style={
                  Object {
                    "color": "#66fcf1",
                  }
                }
              >
                Dark
              </label>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="searchBar"
    >
      <input
        onChange={[Function]}
        placeholder="Search.."
        type="text"
        value=""
      />
    </div>
  </div>
  <List
    currText=""
    pokeType={Array []}
    pokeWeakness={Array []}
  />
</Fragment>
`;

exports[`Navbar Components input first click snapshot 1`] = `
<input
  checked={false}
  name="Grass"
  onChange={[Function]}
  type="checkbox"
  value="Grass"
/>
`;

exports[`Navbar Components input last click snapshot 1`] = `
<input
  checked={false}
  name="Steel"
  onChange={[Function]}
  type="checkbox"
  value="Steel"
/>
`;

exports[`Navbar Components input search snapshot 1`] = `
<input
  onChange={[Function]}
  placeholder="Search.."
  type="text"
  value=""
/>
`;
